lib/logstash/filters/grok.rb

```diff
@@ -191,7 +193,16 @@ class LogStash::Filters::Grok < LogStash::Filters::Base
 
     # Append values to the `tags` field when there has been no
     # successful match
-    config :tag_on_failure, :validate => :array, :default => ["_grokparsefailure"]
+    config :tag_on_failure, :validate => :array, :default => "_grokparsefailure"
+
+    # Attempt to terminate regexps after this amount of time.
+    # This applies per pattern if multiple patterns are applied
+    # This will never timeout early, but may take a little longer to timeout.
+    # Actual timeout is approximate based on a 250ms quantization.
+    config :timeout_millis, :validate => :number, :default => 2000
+
+    # Tag to apply if a grok regexp tims out.
+    config :tag_on_timeout, :validate => :string, :default => '_groktimeout'
```

Comment:
Tags in Logstash have typically been a kind of a set, so my preference is to always treat them as a set where operations on that set are union, and operations usually involve two sets, not a set and a singular value (exception: add_tag).. At least, that's my thought anyway.

I don't have a strong preference. It's fine to have this `:validate => :string`, probably.